sound/soc/sof/ipc4-topology.c

```diff
@@ -1798,18 +1798,17 @@ sof_ipc4_prepare_copier_module(struct snd_sof_widget *swidget,
 	gtw_cfg_config_length = copier_data->gtw_cfg.config_length * 4;
 	ipc_size = sizeof(*copier_data) + gtw_cfg_config_length;
 
-	if (ipc4_copier->dma_config_tlv.type == SOF_IPC4_GTW_DMA_CONFIG_ID &&
-	    ipc4_copier->dma_config_tlv.length) {
-		dma_config_tlv_size = sizeof(ipc4_copier->dma_config_tlv) +
-			ipc4_copier->dma_config_tlv.dma_config.dma_priv_config_size;
+	dma_config_tlv_size = 0;
+	for (i = 0; i < SOF_IPC4_DMA_DEVICE_MAX_COUNT; i++) {
+		if (ipc4_copier->dma_config_tlv[i].type != SOF_IPC4_GTW_DMA_CONFIG_ID)
+			continue;
+		dma_config_tlv_size += ipc4_copier->dma_config_tlv[i].length;
```

Comment:
not following, you are combining the length of two TLVs? How do you account for the T and L parts for each TLV?

Reply:
> not following, you are combining the length of two TLVs? How do you account for the T and L parts for each TLV?

Sorry @plbossart, I didn't get your question. Does T mean Type and L mean Length? Yes, I am combining the length of two TLVs, but the two TLVs are the same type (SOF_IPC4_GTW_DMA_CONFIG_ID). The purpose here is to calculate the total size of all used TLVs and append the size to ipc_size.

Reply:
When you have a TLV, the length L is usually the size of the V.

so if you want modify the size it should be (sizeof(T) + sizeof(L) + L), no?

Reply:
> When you have a TLV, the length L is usually the size of the V.
> 
> so if you want modify the size it should be (sizeof(T) + sizeof(L) + L), no?

Thanks @plbossart. I got it now.